topics/ui/controls/built_in_button.md

```diff
@@ -79,44 +68,74 @@ extComboBox.setEditor(new BasicComboBoxEditor() {
 
 </tab>
 </tabs>
+</chapter>
 
-Do **not** place the button on the right of the control.
 
-![](browse_buttons.png){width=250}
+### Expand a field
 
-### Expand field
-
-If the input text can be long and place is constrained, use a built-in button to expand the control
+If the input text can be long and the place is constrained,
+use the **Expand** button to expand the control
 ([`ExpandableTextField`](%gh-ic%/platform/platform-api/src/com/intellij/ui/components/fields/ExpandableTextField.java)):
 
-![](expandable_1.png){width=332}
-
-![](expandable_2.png){width=582}
+![Collapsed built-in button](built_in_button_collapsed.png){width=706}
 
-Do **not** use the Show Viewer button instead.
+When the field is expanded, show the **Collapse** button:
 
-![](input_expand.png){width=357}
+![Expanded built-in button](built_in_button_expanded.png){width=706}
 
 ### List values
 
-Use a control with the table icon to select from the list of classes, methods or environment variables:
+Use the **List** icon to select a value from the list of classes, methods, or environment variables:
 
-![](input_table.png){width=250}
+![List of values](built_in_button_list.png){width=706}
 
-Use a combo box instead of the Variables button. This icon works as a combo box.
+### Add value
+Use the **Add** button to select multiple values, such as files, variables, folder paths. The selected values will be added to the existing list.
 
-![](variables_combobox.png){width=514}
+![Add a value to the field](built_in_button_add.png){width=706}
 
-### Add value
+## How to use
 
-The Plus button works the same way as the Browse button.
-The only difference is that the selected value is added, instead of overwriting the existing one.
-Place the plus icon inside the control.
-![](plus.png){width=250}
+Place the built-in button inside the input field. Do not place the built-in button next to the field control:
+<table style="none" border="false">
+<tr>
+<td><format color="369650" style="bold">Correct</format>
+<img src="built_in_button_browse_correct.png" alt="Browse button" width="378"/></td>
+<td><format color="E55765" style="bold">Incorrect</format>
+<img src="built_in_button_browse_incorrect.png" alt="Incorrect browse button" width="378"/></td>
+</tr>
+</table>
 
-### Copy, Info
+To place a button inside a text field, use [`ExtendableTextField`](%gh-ic%/platform/platform-api/src/com/intellij/ui/components/fields/ExtendableTextField.java) and
+its `addExtension()` method.
```

Comment:
Possibly this part should be in the "Implementation" box at the top like it is for the Split Button?